### ISSUE TITLE
Add an option for updating in areas where disclosing HTTPS Everywhere use may be risky.

### ISSUE THREAD

**Issue by ghost** [EFForg/https-everywhere#15208]
Type: feature request

cc @Hainish @jeremyn.

HTTPS Everywhere updates have a easily detectable fingerprint, like DNS queries to `https-rulesets.org` and HTTPS requests to Fastly, which is virtually unheard of in (for example) [Russia](https://www.google.com/search?q=fastly+cdn&lr=lang_ru) and [Eastern Europe](https://www.google.com/search?q=fastly+cdn&lr=lang_uk) (these countries also have low human rights ratings) and in some areas may get a person flagged for lifelong surveillance or cause a visit by the police as a suspected foreign journalist, activist or otherwise someone who has something to hide. In addition, [some countries may block entire CDNs](https://www.bleepingcomputer.com/news/government/russia-bans-18-million-amazon-and-google-ips-in-attempt-to-block-telegram/).

I suggest using multiple content delivery networks (at least two), removing the obvious `https-rulesets.org` domain name that could be used as an anchor for surveillance software (it could be kept for direct ruleset downloads), and allow the user to choose the CDN, with a preferred CDN being chosen automatically, with automatic fallback if a CDN is not available.

I know the obvious answer is to use a VPN to hide the update fingerprint or access the blocked CDN, but in some areas [using a VPN is a punishable offense](https://thebestvpn.com/are-vpns-legal-banned-countries/) or may be suspicious or otherwise risky.

**Comment by Bisaloo** [comment]
See https://github.com/EFForg/https-everywhere/issues/12606#issuecomment-330818029 for previous discussions on this topic.

**Comment by epicminecrafting** [comment]
@Bisaloo I don't think this is a duplicate, but thanks for the link.

**Comment by sazeem15** [comment]
So your are suggesting that using this feature could risk our privacy of losing information on open internet. As far as my understanding what ever we search or browse for on internet using or under this protocol Https everywhere, we are losing our privacy. correct me if i am wrong?
Many Thanks

**Comment by YegorIevlev** [comment]
@Giltyhub That's even more suspicious than what is currently being used.

**Comment by Giltyhub** [comment]
@YegorIevlev From a traffic analysis point of view I don't think encrypted ipfs p2p traffic will immediately tell that someone is trying to get https-rulesets updates.

**Comment by epicminecrafting** [comment]
@Hainish Can you at least remove all mentions of `eff.org` from the `https-rulesets.org` certificate? That will protect the users from adversaries looking to punish them for visiting "politically inappropriate" websites like `eff.org`. The current certificate may be confused with EFF website's certificate and could lead to prosecution of the users of HTTPS Everywhere by attackers misled by `eff.org` in the certificate.

**Comment by jeremyn** [comment]
@epicminecrafting The relationship between `https-rulesets.org` and the EFF is public, for example see the main page https://www.https-rulesets.org/ . This helps people who see traffic from their computer to this domain know what that traffic is for. Even if references to HTTPS Everywhere or the EFF were removed from https://www.https-rulesets.org/ , the SSL certificate changed, and whatever else, the relationship is still public knowledge (such as through this repository), discoverable by the technical security agency of a hostile government who could add it to their automated screening for "politically inappropriate" traffic.

In other words removing the connection to the EFF from the `https-rulesets.org` site would interfere with a legitimate use case, but is only superficial protection against the risk I think you're describing.

For what it's worth I pointed out that risk in an earlier discussion about this feature, in issue https://github.com/EFForg/https-everywhere/issues/12606, for example see my comment at https://github.com/EFForg/https-everywhere/issues/12606#issuecomment-331319374 and following.
